test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_full_dispatch.cpp

```diff
@@ -0,0 +1,367 @@
+                               cl_command_queue queue)
+        : InfoMutableCommandBufferTest(device, context, queue),
+          svm_buffers(context), group_size(0)
+    {}
+
+    bool Skip() override
+    {
+        cl_mutable_dispatch_fields_khr mutable_capabilities;
+        cl_mutable_dispatch_fields_khr requested =
+            CL_MUTABLE_DISPATCH_GLOBAL_OFFSET_KHR
+            | CL_MUTABLE_DISPATCH_GLOBAL_SIZE_KHR
+            | CL_MUTABLE_DISPATCH_LOCAL_SIZE_KHR
+            | CL_MUTABLE_DISPATCH_ARGUMENTS_KHR
+            | CL_MUTABLE_DISPATCH_EXEC_INFO_KHR;
+        bool mutable_support =
+            !clGetDeviceInfo(
+                device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+                sizeof(mutable_capabilities), &mutable_capabilities, nullptr)
+            && (mutable_capabilities & requested) == requested;
+
+        cl_device_svm_capabilities svm_caps;
+        bool svm_capabilities =
+            !clGetDeviceInfo(device, CL_DEVICE_SVM_CAPABILITIES,
+                             sizeof(svm_caps), &svm_caps, NULL)
+            && svm_caps != 0;
+
+        return !svm_capabilities || !mutable_support
+            || InfoMutableCommandBufferTest::Skip();
+    }
+
+    // setup kernel program specific for command buffer with full mutable
+    // dispatch test
+    cl_int SetUpKernel() override
+    {
+        const char *kernel_str =
+            R"(typedef struct {
+                global int* ptr;
+            } wrapper;
+            __kernel void full_dispatch(__global int *src, __global wrapper *dst)
+            {
+                size_t gid = get_global_id(0) % get_global_size(0);
+                size_t lid = gid % get_local_size(0);
+                dst->ptr[gid] = src[lid];
+            })";
+
+        cl_int error = CL_SUCCESS;
+
+        error = create_single_kernel_helper_create_program(context, &program, 1,
+                                                           &kernel_str);
+        test_error(error, "Failed to create program with source");
+
+        error = clBuildProgram(program, 1, &device, nullptr, nullptr, nullptr);
+        test_error(error, "Failed to build program");
+
+        kernel = clCreateKernel(program, "full_dispatch", &error);
+        test_error(error, "Failed to create copy kernel");
+
+        return CL_SUCCESS;
+    }
+
+    // setup kernel arguments specific for command buffer with full mutable
+    // dispatch test
+    cl_int SetUpKernelArgs() override
+    {
+        // query max work-group size needed for allocation size of input buffers
+        size_t workgroupinfo_size = 0;
+        cl_int error = clGetKernelWorkGroupInfo(
+            kernel, device, CL_KERNEL_WORK_GROUP_SIZE, sizeof(group_size),
+            &workgroupinfo_size, NULL);
+        test_error(error, "clGetKernelWorkGroupInfo failed");
+
+        group_size = std::min(num_elements, workgroupinfo_size);
+        const size_t sizeToAllocateSrc = group_size * sizeof(cl_int);
```

Comment:
`sizeToAllocateSrc` - nitpick that other variables are snake_case

Reply:
Done